### PR TITLE
Move `torch.linalg` opinfos to opinfo.definitions (1/2)

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83554
* __->__ #83547
* #83563
* #83560
* #83559

### CODE REVIEW

File: torch/testing/_internal/opinfo/definitions/linalg.py [hunk 1]
```diff
@@ -0,0 +1,1180 @@
...
+    This function generates input for torch.linalg.householder_product (torch.orgqr).
+    The first argument should be a square matrix or batch of square matrices, the second argument is a vector or batch of vectors.
+    Empty, square, rectangular, batched square and batched rectangular input is generated.
+    """
+    # Each column of the matrix is getting multiplied many times leading to very large values for
+    # the Jacobian matrix entries and making the finite-difference result of grad check less accurate.
+    # That's why gradcheck with the default range [-9, 9] fails and [-2, 2] is used here.
+    samples = (
+        SampleInput(
+            make_tensor(
```
Comment (albanD) [inline torch/testing/_internal/opinfo/definitions/linalg.py:124]: that formatting takes a lot of space...

**Review by albanD — APPROVED** [review]

File: torch/testing/_internal/opinfo/definitions/linalg.py (hunk 1 above)
Comment (peterbell10) [reply torch/testing/_internal/opinfo/definitions/linalg.py:124]: Very true. IIRC the reasoning is that adding a new element in the middle of a list shouldn't require the entire list to be re-wrapped and therefore show up as changed in the diff. I've definitely seen lots of cases where merge conflicts could have been avoided if we used this style uniformly. So I think it's a reasonable trade-off.

Comment (albanD) [reply torch/testing/_internal/opinfo/definitions/linalg.py:124]: Ok!

**Pull request closed without merging**